doc/neps/nep-0028-deprecation_policy.rst

```diff
@@ -0,0 +1,184 @@
+
+
+   - support minor versions of ``CPython`` initially released
+     42 months prior to our planned release date
+   - always support at least the 2 latest versions of ``CPython``
+   - support minor versions of ``numpy`` initially released in the 24
+     months prior to our planned release date or oldest that supports the
+     minimum Python version (whichever is higher)
+
+   The minimum supported version of ``CPython`` will be set to
+   ``python_requires`` in ``setup`` and all supported versions of
+   Python will be in the test matrix and have binary artifacts built
+   for releases.
+
+   We will bump the minimum Python and NumPy versions as we can on
+   every minor and major release, but never on a patch release.
+
+For other dependencies, adopt similar time windows 24 months shorter.
+
+
+Backward compatibility
+----------------------
+
+No issues other than the intentional dropping of old version of
+``CPython`` and dependencies.
+
+
+Alternatives
+------------
+
+Ad-Hoc
+~~~~~~
+
+Projects could on every release evaluate if they want to increase
+the minimum version of Python supported.  While this is a notionally
+simple policy, it makes it hard for downstream users to predict what
+the future minimum versions will be.  As there is no objective threshold
+to when the minimum version should be dropped, it is easy for these
+discussions to devolve into bike shedding and acrimony.
+
+
+All Python Software Foundation supported versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a very clear and conservative approach.  However, it means that
+there is 4 year lag between when new language features come into the
+language and when the projects are able to use them.  Additionally,
+for projects that have a significant component of compiled extensions
+this requires building many binary artifacts for each release.
+
+For the case of NumPy, many projects carry workarounds to bugs that
+are fixed in subsequent versions of NumPy.  Being proactive about
+increasing the minimum version of NumPy will allow downstream
+packages to carry fewer version-specific patches.
+
+
+
+Default version on Linux distribution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The policy could be support the version of Python that ships by
+default in the latest Ubuntu LTS or CentOS/RHEL release.  However, we
+would still have to standardize across the community which
+distribution we are following.
+
+By following the versions supported by major Linux distributions, we
+are giving up technical control of our projects to external
+organizations that may have different motivations and concerns than we
+do.
+
+N minor versions of Python
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the current release cadence of the CPython, the proposed time
+(42 months) is roughly equivalent to "the last two" CPython minor
+versions.  However, if CPython changes their release cadence, any rule
+based on the number of minor releases will need to be changed.
```

Comment:
But isn't part of the rule being suggested based on the number of minor releases?
> always support at least the 2 latest versions of ``CPython``

By version, do you not mean "minor release" e.g. 3.6 and 3.7? 
Sorry if I'm misunderstanding - but if I do, maybe others would?

Reply:
The rule is based on time (42 month, 3 and a half year), which happen to be 2 minor version of (C)Python with current release cadence. As pointed by Carol, the release cadence of Python is likely to go up so that would be 3 or  minor releases if the new PEP is accepted.

Reply:
> The rule is based on time (42 month, 3 and a half year), which happen to be 2 minor version 

That's what I thought, as the document is clear as to where the 42 month number comes from. In that case, I would suggest "always support at least the 2 latest versions of ``CPython``" should be dropped from the recommended language for project development guidelines, as it is redundant. If it should be left in for clarification of the 42 month time-based rule, maybe it should be explained below the bullets so it does not look like a separate rule.

Reply:
This is mostly as a hedge in case anything goes sideways upstream and we don't have a Python release for 42 months.  In that case we may have bigger problems.....